app/api/util.spec.ts

```diff
@@ -136,3 +136,22 @@ describe('synthesizeData', () => {
     ])
   })
 })
+
+test('instanceCan', () => {
+  expect(instanceCan.start({ runState: 'running' })).toBe(false)
+  expect(instanceCan.start({ runState: 'stopped' })).toBe(true)
+
+  // @ts-expect-error typechecker rejects actions that don't exist
+  // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+  instanceCan.abc
+})
+
+test('diskCan', () => {
+  expect(diskCan.delete({ state: { state: 'creating' } })).toBe(false)
+  expect(diskCan.delete({ state: { state: 'attached', instance: 'xyz' } })).toBe(false)
+  expect(diskCan.delete({ state: { state: 'detached' } })).toBe(true)
+
+  // @ts-expect-error typechecker rejects actions that don't exist
+  // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+  diskCan.abc
+})
```

Comment:
Confirmed the `@ts-expect-error`s fail on main:

![Screenshot 2024-10-15 at 2 03 47 PM](https://github.com/user-attachments/assets/a342d337-4134-48c6-b284-2d1705a0b06c)